other/python_transformations/interface_builder.py

```diff
@@ -0,0 +1,395 @@
+            gap (float, optional): gap between film and substrate. Defaults to 2.0.
+            vacuum_over_film (float, optional): vacuum over the top of the film. Defaults to 20.0.
+            film_thickness (float, optional): the film thickness. Defaults to 1.
+            substrate_thickness (float, optional): substrate thickness. Defaults to 1.
+            in_layers (bool, optional): set the thickness in layer units. Defaults to True.
+
+        Yields:
+            Iterator[Interface]: interfaces from slabs
+        """
+        film_sg = SlabGenerator(
+            self.film_structure,
+            self.film_miller,
+            min_slab_size=film_thickness,
+            min_vacuum_size=3,
+            in_unit_planes=in_layers,
+            center_slab=True,
+            primitive=True,
+            reorient_lattice=False,  # This is necessary to not screw up the lattice
+        )
+
+        sub_sg = SlabGenerator(
+            self.substrate_structure,
+            self.substrate_miller,
+            min_slab_size=substrate_thickness,
+            min_vacuum_size=3,
+            in_unit_planes=in_layers,
+            center_slab=True,
+            primitive=True,
+            reorient_lattice=False,  # This is necessary to not screw up the lattice
+        )
+
+        film_shift, sub_shift = self._terminations[termination]
+
+        film_slab = film_sg.get_slab(shift=film_shift)
+        sub_slab = sub_sg.get_slab(shift=sub_shift)
+
+        for match in self.zsl_matches:
+            # Build film superlattice
+            super_film_transform = np.round(
+                from_2d_to_3d(get_2d_transform(film_slab.lattice.matrix[:2], match.film_sl_vectors))
+            ).astype(int)
+            film_sl_slab = film_slab.copy()
+            film_sl_slab.make_supercell(super_film_transform)
+            assert_allclose(
+                film_sl_slab.lattice.matrix[2], film_slab.lattice.matrix[2], atol=1e-08
+            ), "2D transformation affected C-axis for Film transformation"
+            assert_allclose(
+                film_sl_slab.lattice.matrix[:2], match.film_sl_vectors, atol=1e-08
+            ), "Transformation didn't make proper supercell for film"
+
+            # Build substrate superlattice
+            super_sub_transform = np.round(
+                from_2d_to_3d(get_2d_transform(sub_slab.lattice.matrix[:2], match.substrate_sl_vectors))
+            ).astype(int)
+            sub_sl_slab = sub_slab.copy()
+            sub_sl_slab.make_supercell(super_sub_transform)
+            assert_allclose(
+                sub_sl_slab.lattice.matrix[2], sub_slab.lattice.matrix[2], atol=1e-08
+            ), "2D transformation affected C-axis for Film transformation"
+            assert_allclose(
+                sub_sl_slab.lattice.matrix[:2], match.substrate_sl_vectors, atol=1e-08
+            ), "Transformation didn't make proper supercell for substrate"
+
+            # Add extra info
```

Comment:
Comment doesn't tell me much. What's in the extra info? Why do we ignore keys that start with '@'?